src/components/logo/logo.tsx

```diff
@@ -27,14 +30,60 @@ const mapStateToProps = state => ({
  */
 @connect(mapStateToProps)
 @withPlayer
+@withEventManager
 @withLogger(COMPONENT_NAME)
 @withText({logoText: 'controls.logo'})
 class Logo extends Component<any, any> {
+  private _logoRef: RefObject<HTMLAnchorElement> = createRef<HTMLAnchorElement>();
+
+  /**
+   * before component mounted, set initial state of the logo
+   *
+   * @returns {void}
+   * @memberof Logo
+   */
+  public componentWillMount(): void {
+    this.setState({isUrlClickable: true});
+  }
+
+  /**
+   * when component did mount
+   *
+   * @returns {void}
+   * @memberof Logo
+   */
+  public componentDidMount(): void {
+    this._handleLogoUrl();
+  }
+
+  /**
+   * handles the logo url
+   * if the url contains ${entryId}, then replace it with the played entry id
+   *
+   * @returns {void}
+   * @memberof Logo
+   */
+  private _handleLogoUrl(): void {
+    const url = this.props.config.url;
+    if (url && url.indexOf(ENTRY_VAR) !== -1) {
+      const {player, eventManager} = this.props;
+      eventManager.listen(player, player.Event.MEDIA_LOADED, () => {
+        const entryId = player.sources.id;
```

Comment:
Player.sourcs.id is planand to be  eliminated 
And should not be accessible 
Entry Id should be fetched fromt the public API

Reply:
`sources` is a public API of kaltura player instance.